src/test/java/com/prgms/springbootbasic/persistent/VouchersInDBTest.java

```diff
@@ -0,0 +1,168 @@
+
+import com.prgms.springbootbasic.domain.FixedAmountVoucher;
+import com.prgms.springbootbasic.domain.PercentAmountVoucher;
+import com.prgms.springbootbasic.domain.Voucher;
+import com.prgms.springbootbasic.util.YamlPropertyFactory;
+import com.zaxxer.hikari.HikariDataSource;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.sql.DataSource;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.*;
+
+@SpringJUnitConfig
+class VouchersInDBTest {
+
+    @Configuration
+    @ComponentScan(basePackages = "com.prgms.springbootbasic")
+    @PropertySource(value = "application.yaml", factory = YamlPropertyFactory.class)
+    static class TestConfig {
+
+        @Bean
+        public DataSource dataSource(@Value("${database.url}") String url,
+                                     @Value("${database.hostname}") String hostname,
+                                     @Value("${database.password}") String password) {
+            HikariDataSource datasource = DataSourceBuilder.create()
+                    .url(url)
+                    .username(hostname)
+                    .password(password)
+                    .type(HikariDataSource.class)
+                    .build();
+            return datasource;
+        }
+
+        @Bean
+        public JdbcTemplate jdbcTemplate(DataSource dataSource) {
+            return new JdbcTemplate(dataSource);
+        }
+
+        @Bean
+        public NamedParameterJdbcTemplate namedParameterJdbcTemplate(JdbcTemplate jdbcTemplate) {
+            return new NamedParameterJdbcTemplate(jdbcTemplate);
+        }
+
+    }
+
+    @Autowired
+    private VouchersInDB vouchersInDB;
+
+    @Test
+    @DisplayName("바우처가 저장된다.")
+    void 바우처_저장() {
+        Voucher voucher = new FixedAmountVoucher(UUID.randomUUID(), 1000L);
+        vouchersInDB.save(voucher);
+        Voucher findVoucher = vouchersInDB.findById(voucher.getVoucherId());
+        // 필드 이름 모두 같은지 확인
+        assertThat(voucher).usingRecursiveComparison()
+                .isEqualTo(voucher);
```

Comment:
```suggestion
        Voucher voucher = new FixedAmountVoucher(UUID.randomUUID(), 1000L);
        vouchersInDB.save(voucher);
        Voucher findVoucher = vouchersInDB.findById(voucher.getVoucherId());
        
        assertThat(findVoucher).usingRecursiveComparison()
                .isEqualTo(voucher);
```

또 findVoucher 라는 네이밍도 조금 신경써주시면 좋겠습니다!